Neos.Fusion/Migrations/Code/Version20220326120900.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Core\Migrations;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Expand Neos.Fusion-FusionObjects without namespace to fully qualified names
+ */
+class Version20220326120900 extends AbstractMigration
+{
+    /**
+     * The name of all prototypes from Neos.Fusion that could be used without
+     * namespace previously because of the `default` namespace
+     */
+    protected array $namesToMigrate = [
+        'Array',
+        'RawArray',
+        'Join',
+        'DataStructure',
+        'Template',
+        'Case',
+        'Matcher',
+        'Renderer',
+        'Value',
+        'Component',
+        'CanRender',
+        'DebugDump',
+        'Debug',
+        'Collection',
+        'RawCollection',
+        'Loop',
+        'Map',
+        'Reduce',
+        'Http.ResponseHead',
+        'Http.Message',
+        'Attributes',
+        'Tag',
+        'UriBuilder',
+        'Augmenter',
+        'ResourceUri',
+        'Link.Action',
+        'Link.Resource',
+        'Fragment',
+        'GlobalCacheIdentifiers'
+    ];
+
+    public function getIdentifier():string
+    {
+        return 'Neos.Fusion-20220326120900';
+    }
+
+    public function up():void
+    {
+        foreach ($this->namesToMigrate as $name) {
+            // fusion object assignments
+            $this->searchAndReplaceRegex('/(?<=\\=)([\\s]*)(' . preg_quote($name) . ')(?=[$\\s\\{])/u', '$1Neos.Fusion:$2', ['fusion']);
```

Comment:
I didn't test, but my PhpStorm complaints about a "redundant character escape" and encourages to specify the delimiter parameter for `preg_quote`:
```suggestion
            $this->searchAndReplaceRegex('/(?<==)([\\s]*)(' . preg_quote($name, '/') . ')(?=[$\\s{])/u', '$1Neos.Fusion:$2', ['fusion']);
```

Reply:
jup the escape is indeed redundant in this case as the $names dont contain fancy chars - but i would keep the preg_quote as i thinkg it is good practice to escape those inputs first ...

Reply:
@bwaidelich just took the regex from the old migration for 4.0 because that one caused no harm in the past. 

Unless you see problems that can be caused by the regex would rather use the flight proven code.

Reply:
> i would keep the preg_quote

Of course, the `preg_quote()` is required. But for it to work reliably, the 2nd parameter should be specified usually.

> Unless you see problems that can be caused by the regex would rather use the flight proven code

Sure. It's just the double back slashes that don't have an effect. But they won't do harm.
Also the `preg_quote()` call isn't dangerous in this case since we know `$name`